Limited Global Styles: Open launchbar popup by default

mmtr opened the pull request:
Fixes https://github.com/Automattic/dotcom-forge/issues/2268

## Proposed Changes

Improves the Limited Global Styles notice displayed in the launchbar:
- The dialog is open by default when custom styles are previewed, so it's less likely to miss.
- Wording has been adapted to make the UX more intuitive.

State | Before | After
--- | --- | ---
Preview ON | <img width="959" alt="Screenshot 2023-05-01 at 15 20 00" src="https://user-images.githubusercontent.com/1233880/235458472-67044f6c-237a-432a-88a1-b6bc0f85922e.png"> | <img width="994" alt="Screenshot 2023-05-01 at 15 17 55" src="https://user-images.githubusercontent.com/1233880/235458306-ff676b8d-e3be-4e3c-b23b-e30824d20681.png">
Preview ON (mobile) | <img width="409" alt="Screenshot 2023-05-01 at 15 19 40" src="https://user-images.githubusercontent.com/1233880/235458664-e98e322f-55dd-48e0-83e8-fbeed5ef8aaf.png"> | <img width="401" alt="Screenshot 2023-05-01 at 15 18 44" src="https://user-images.githubusercontent.com/1233880/235458904-87a2c093-16e5-4bab-bad1-4c1c736c7de0.png">
Preview OFF | <img width="962" alt="Screenshot 2023-05-01 at 15 20 27" src="https://user-images.githubusercontent.com/1233880/235458692-4cb17d1c-fb94-4cff-a0ab-0e30229aa5e6.png"> | <img width="996" alt="Screenshot 2023-05-01 at 15 18 14" src="https://user-images.githubusercontent.com/1233880/235458736-499bce2d-8e3f-41cc-8b1a-5342c3a89bc3.png">
Preview OFF (mobile) | <img width="416" alt="Screenshot 2023-05-01 at 15 19 32" src="https://user-images.githubusercontent.com/1233880/235458775-69c35f2a-42b3-4208-ad33-cb09f6b94d6f.png"> | <img width="421" alt="Screenshot 2023-05-01 at 15 18 32" src="https://user-images.githubusercontent.com/1233880/235458937-420cff34-5cc4-4552-8227-5347ac172ec7.png">

## Testing Instructions

- Apply these changes to your sandbox.
- Sandbox a free site that uses Global Styles.
- Visit the frontend.
- Make sure the dialog is open by default.
- Make sure you can show the custom styles using the secondary button in the dialog.
- Make sure the wording is clear.

mmtr committed:
Limited Global Styles: Launchbar notice improvements

mmtr committed:
Better copy

BogdanUngureanu commented:
Great job, Miguel, this looks really good. I've noticed some small issues though I think we should fix them before shipping it:

- On mobile resolutions, clicking on the modal text will redirect the users to the plans page - which is an unexpected behavior.
- The modal remains open between refreshes even though I clicked it on the panel to collapse it. This might be annoying to users.
- When the styles are hidden, the modal is always collapsed (inconsistent with the other state where is always open).
-  When clicking on the upgrade and on the Turn on/off preview buttons we aren't triggering the events we used to trigger.
- I think we should add a new event that tracks the impressions for the modal. This would be inline with the impression events we trigger for the other GS nudges.

mmtr committed:
Fix events

mmtr committed:
Track whether the dialog is shown/hidden

mmtr committed:
Track initial dialog impression

mmtr committed:
Keep dialog hidden once it's collapsed

mmtr commented:
> On mobile resolutions, clicking on the modal text will redirect the users to the plans page - which is an unexpected behavior.

I cannot replicate this – Am I missing any testing step?

https://user-images.githubusercontent.com/1233880/235696356-5dd25571-916f-42f2-ac9d-23a64663735f.mov

> The modal remains open between refreshes even though I clicked it on the panel to collapse it.

Solved in https://github.com/Automattic/wp-calypso/pull/76459/commits/58452f9676d97c8238515c556d34ab4bbaac331d

> When the styles are hidden, the modal is always collapsed (inconsistent with the other state where is always open).

Yes, this is intentional. The reason why the dialog is open by default is to provide awareness to users and ensure it's less likely to miss. If users turned off the preview, they are already aware of the notice, so we no longer need to display the dialog open by default.

> When clicking on the upgrade and on the Turn on/off preview buttons we aren't triggering the events we used to trigger

Good catch! Fixed in https://github.com/Automattic/wp-calypso/pull/76459/commits/a2bcb8eedd3b5d5a0224444cfa91e71429bad874

> I think we should add a new event that tracks the impressions for the modal

Added a new `action` property in https://github.com/Automattic/wp-calypso/pull/76459/commits/3d384095f4ac0d3587df8d0130244f016fb15c39 and https://github.com/Automattic/wp-calypso/pull/76459/commits/61c79814c3b32477ea4ba9cf4d9a63d4da14952b to the `wpcom_global_styles_gating_notice` event so we can determine whether the dialog has been displayed.

mmtr committed:
Simplify logic

BogdanUngureanu commented on apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/wpcom-global-styles-view.js:
```diff
@@ -23,13 +24,25 @@ document.addEventListener( 'DOMContentLoaded', () => {
+			action: popover?.classList.contains( 'hidden' ) ? 'show' : 'hide',
+		} );
 		popover?.classList.toggle( 'hidden' );
+		localStorage.setItem( 'showLimitedGlobalStylesNotice', '0' );
```
Maybe the value should be toggled instead of always being persisted as '0'? If the user leaves it open, then it should remain opened? 🤔

mmtr replied:
https://github.com/Automattic/wp-calypso/pull/76459/commits/88331cd28674f165f5402fe409622f2d38de87e1

BogdanUngureanu commented:
> I cannot replicate this – Am I missing any testing step?

Looks like I'm not able to reproduce it anymore. 

Looks much better, thanks for addressing the issues. There's one more thing (not blocking). When a user clicks on the Turn on/off preview button we log an event. However, we can't differentiate between the Turn on and the Turn off actions. What do you think if we add a new prop as we did for the modal collapse? :) 

I've also added a small comment about the modal collapse state.

BogdanUngureanu commented:
Non-blocking: On mobile view, "Upgrade" label might not be explicit enough, especially on closed modal states. But again, no alternative comes to mind. It's an improvement anyway. 

Non-blocking: Are we sure that having the modal open on mobile viewports is clear enough on how to close it? As a user, when visiting the front-end for the first time, it might be not intuitive enough.

mmtr committed:
Leave default preview mode change for a follow-up

mmtr committed:
Keep collapsed status between reloads

mmtr committed:
Add close icon

mmtr commented:
> Are we sure that having the modal open on mobile viewports is clear enough on how to close it?

Added a close button in https://github.com/Automattic/wp-calypso/pull/76459/commits/fc231a2e4d4054910c4a6538a560a7590959c98a so users have a more natural way of closing it.

davewhitley commented:
Current specs (pending design feedback) can be found here: https://www.figma.com/file/PgUzt3AFZ6v6rWrmXO1uDl/Limited-Global-Styles-(upsells)?type=design&node-id=783-7960&t=P0uDQIXKwpcVMTIa-4

mmtr committed:
Design tweaks

mmtr committed:
Merge remote-tracking branch 'origin/trunk' into update/limit-gs-launchbar

BogdanUngureanu reviewed (approved):
LGTM! 🚢

mmtr commented:
@davewhitley here you are!
<img width="427" alt="Screenshot 2023-05-04 at 17 44 50" src="https://user-images.githubusercontent.com/1233880/236259888-73c1779d-de2a-469a-98ae-2480fd15549e.png">
